qiling/loader/elf.py

```diff
@@ -92,13 +92,9 @@ def run(self):
         elffile = ELFFile(fstream)
         elftype = elffile['e_type']
 
-        stack_perm = UC_PROT_NONE
-        for seg in elffile.iter_segments('PT_GNU_STACK'):
-            stack_perm = QlLoaderELF.seg_perm_to_uc_prot(seg['p_flags'])
-
-        QlLoaderELF.seg_perm_to_uc_prot(stack_perm)
-
         # setup program stack
+        stack_seg = elffile.iter_segments('PT_GNU_STACK')
+        stack_perm = QlLoaderELF.seg_perm_to_uc_prot(next(stack_seg)['p_flags'])
```

Comment:
What if there is no such segment in the file? The code should be prepared to handle this situation (that is, default to RWX).